File: common-application/src/main/java/io/quartic/common/healthcheck/PingPongHealthCheck.java
package io.quartic.common.healthcheck;

import com.codahale.metrics.health.HealthCheck;
import io.quartic.common.client.ClientBuilder;
import io.quartic.common.pingpong.PingPongService;

public class PingPongHealthCheck extends HealthCheck {
    private final PingPongService pingPong;

    public PingPongHealthCheck(Class<?> owner, String url) {
        this.pingPong = ClientBuilder.build(PingPongService.class, owner, url);
    }

    @Override
    protected Result check() throws Exception {
        return Result.healthy(pingPong.ping().version());
    }
}

File: common-application/src/main/kotlin/io/quartic/common/healthcheck/PingPongHealthCheck.kt
package io.quartic.common.healthcheck

import com.codahale.metrics.health.HealthCheck
import com.codahale.metrics.health.HealthCheck.Result.healthy
import io.quartic.common.client.ClientBuilder
import io.quartic.common.pingpong.PingPongClient
import java.net.URI

class PingPongHealthCheck(clientBuilder: ClientBuilder, url: URI) : HealthCheck() {
    private val pingPong: PingPongClient = clientBuilder.retrofit(url)

    override fun check(): Result = healthy(pingPong.pingAsync().get().version)
}
